### PR TITLE
Fix display of binary name in error messages.

### PR DESCRIPTION
**Pull request by sellout**
Ensures that test failures report the name of the actual binary that was run.

### CODE REVIEW

**Review by str4d — DISMISSED** [review]
utACK

File: qa/rpc-tests/test_framework/util.py [hunk 1]
```diff
@@ -54,6 +54,9 @@
 PORT_RANGE = 5000
 
 
+def bitcoind_binary():
```
Comment (daira) [inline qa/rpc-tests/test_framework/util.py:57]:
```suggestion
def zcashd_binary():
```

**Comment by daira** [comment]
Test failures need fixing.

**Review by nuttycom — DISMISSED** [review]
utACK

File: qa/rpc-tests/test_framework/util.py [hunk 2]
```diff
@@ -267,7 +270,7 @@ def rebuild_cache():
                 args.append("-connect=127.0.0.1:"+str(p2p_port(0)))
             bitcoind_processes[i] = subprocess.Popen(args)
             if os.getenv("PYTHON_DEBUG", ""):
-                print("initialize_chain: bitcoind started, waiting for RPC to come up")
+                print("initialize_chain: %s started, waiting for RPC to come up" % zcashd_binary())
```
Comment (daira) [inline qa/rpc-tests/test_framework/util.py:273]:
```suggestion
                print("initialize_chain: %s started, waiting for RPC to come up" % (zcashd_binary(),))
```
Style nit: always using a tuple as the right argument of `%` leads to more regular behaviour that is less likely to hide type errors. (`%` with an untupled value can blow up at runtime if the argument turns out to be a tuple with more elements than there are placeholders in the format string.)

File: qa/rpc-tests/test_framework/util.py [hunk 3]
```diff
@@ -430,17 +433,17 @@ def assert_start_raises_init_error(i, dirname, extra_args=None, expected_msg=Non
...
+            assert ("%s exited" % zcashd_binary()) in str(e) #node must have shutdown
             if expected_msg is not None:
                 log_stderr.seek(0)
                 stderr = log_stderr.read().decode('utf-8')
                 if expected_msg not in stderr:
                     raise AssertionError("Expected error \"" + expected_msg + "\" not found in:\n" + stderr)
         else:
             if expected_msg is None:
-                assert_msg = "bitcoind should have exited with an error"
+                assert_msg = "%s should have exited with an error" % zcashd_binary()
```
Comment (daira) [inline qa/rpc-tests/test_framework/util.py:444]:
```suggestion
                assert_msg = "%s should have exited with an error" % (zcashd_binary(),)
```

**Review by daira — DISMISSED** [review]
utACK with non-blocking suggestions.

File: qa/rpc-tests/test_framework/util.py [hunk 4]
```diff
@@ -430,17 +433,17 @@ def assert_start_raises_init_error(i, dirname, extra_args=None, expected_msg=Non
             node = start_node(i, dirname, extra_args, stderr=log_stderr)
             stop_node(node, i)
         except Exception as e:
-            assert 'bitcoind exited' in str(e) #node must have shutdown
+            assert ("%s exited" % zcashd_binary()) in str(e) #node must have shutdown
```
Comment (daira) [inline qa/rpc-tests/test_framework/util.py:436]:
```suggestion
            assert ("%s exited" % (zcashd_binary(),)) in str(e) #node must have shutdown
```

**Review by daira — APPROVED** [review]
utACK

**Pull request merged**